core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Service.java

```diff
@@ -332,7 +334,7 @@ else if (nonce.isPresent() != oauth2Response.getIdToken().isPresent()) {
         if (userinfoUri.isPresent()) {
             // validate access token is trusted by remote userinfo endpoint
             Request request = Request.builder()
-                    .setMethod(POST)
+                    .setMethod(userInfoRequestMethod.orElse(POST))
```

Comment:
Let's simply use GET

Reply:
What about the risk of regression?

I'm good to change to GET, just a little worried it might break for some existing users:  I've seen many custom OAuth enterprise implementations, not always strictly according spec.

Reply:
I think @lukasz-walkiewicz have tested all the IdP we think we support and it was working just fine.

Reply:
No doubt the commercial idp will work with both POST or GET. I see rather the risk within custom built oidc implementations. if their custom oidc implementation only has the POST endpoint and suddenly it's not working anymore after a Trino upgrade. As I said already, for me it's ok to make the change to GET. Will adapt the code later today.

Reply:
We are not aware of any use case like this. Also it sounds very exotic, I would prefer not to support such use cases without hard evidence it is actually needed.